Bootloader merging fails with python3, [ERROR] 'l' format requires -2147483648 <= number <= 2147483647

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure that nothing follows the Issue request type section, all 
   issue details are within the Description section and no changes are made to the 
   template format (as detailed below).

   *************************************************************************************

-->

### Description

Severity: Critical (breaks build with limited workaround), cannot work in CI-automation.

Mbed OS master #31da50e8fa68a2bbe8c92d70b258a5d19a41b5e3 (19.02.2019)
**mbed-cli 1.9.1**
**Python 3.6.1**
**ARM Compiler 6.11 Professional** or **GCC 6.3.1**
**K64F**

mbed-cloud-client-example (internal development version)

Bootloader merging fails with:
```
Merging Regions
  Filling region bootloader with /home/teetak01/devel/mbed-cloud-client-example-internal/tools/mbed-bootloader-k64f-block_device.bin
[ERROR] 'l' format requires -2147483648 <= number <= 2147483647
[mbed] ERROR: "/usr/bin/python3" returned error.
```

This fails two times, and third retry in compilation passes.

```
Merging Regions
  Filling region bootloader with /home/teetak01/devel/mbed-cloud-client-example-internal/tools/mbed-bootloader-k64f-block_device.bin
  Filling region header with ./BUILD/K64F/ARMC6/mbed-cloud-client-example-internal_header.hex
  Filling region application with ./BUILD/K64F/ARMC6/mbed-cloud-client-example-internal_application.bin
Space used after regions merged: 0x6df34
Merging Regions
  Filling region application with ./BUILD/K64F/ARMC6/mbed-cloud-client-example-internal_application.bin
Space used after regions merged: 0x63b34
```

<!--
    Required
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
    Things to consider sharing:
    - What target does this relate to?
    - What toolchain (name + version) are you using?
    - What tools (name + version - is it mbed-cli, online compiler or IDE) are you using?
    - What is the SHA of Mbed OS (git log -n1 --oneline)?
    - Steps to reproduce. (Did you publish code or a test case that exhibits the problem?)
-->


### Issue request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the issue otherwise.)
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Question
    [ ] Enhancement
    [x] Bug

--- a/tools/build_api.py
+++ b/tools/build_api.py
@@ -398,8 +398,9 @@ def _fill_header(region_list, current_region):
             else:
                 ih = intelhex_offset(region_dict[data].filename, offset=region_dict[data].start)
             if subtype.startswith("CRCITT32"):
-                fmt = {"CRCITT32be": ">l", "CRCITT32le": "<l"}[subtype]
-                header.puts(start, struct.pack(fmt, zlib.crc32(ih.tobinarray())))
+                fmt = {"CRCITT32be": ">L", "CRCITT32le": "<L"}[subtype]
+                crc_val = zlib.crc32(ih.tobinarray()) & 0xffffffff
+                header.puts(start, struct.pack(fmt, crc_val))
             elif subtype.startswith("SHA"):
                 if subtype == "SHA256":
                     hash = hashlib.sha256()

